commands/lint.js

```diff
@@ -52,6 +67,52 @@ function api({strict = false, local = false, fix = false, eslintArgs = []} = {})
 	});
 }
 
+function tslintBin(context) {
+	try {
+		resolver.sync('tslint', {basedir: context});
+		return path.join(context, 'node_modules', '.bin', 'tslint');
+	} catch (e) {
+		return 'tslint';
+	}
+}
+
+function shouldTSLint(context) {
+	if (fs.existsSync(path.join(context, 'tslint.json'))) {
+		if (glob.sync('**/*.+(ts|tsx)', globOpts).length > 0) {
+			try {
+				return !spawn.sync(tslintBin(context), ['-v'], {stdio: 'ignore'}).error;
+			} catch (e) {
+				// ignore
+			}
+		}
+	}
+	return false;
+}
+
+function tslint({fix = false} = {}, context) {
+	const args = ['-p', context];
+	if (fix) args.push('--fix');
+
+	return new Promise((resolve, reject) => {
+		const opts = {env: process.env, cwd: process.cwd(), stdio: 'inherit'};
+		const child = spawn(tslintBin(context), args, opts);
+		child.on('close', code => {
+			if (code !== 0) {
+				reject();
+			} else {
+				resolve();
+			}
+		});
+	});
+}
+
+function api(opts) {
+	const context = packageRoot().path;
+	return Promise.resolve()
+		.then(() => shouldESLint() && eslint(opts))
+		.then(() => shouldTSLint(context) && tslint(opts, context));
```

Comment:
Maybe silly, but should we throw a message here if there aren't any source files at all?

Reply:
Logically, I'd imagine no files = no linting errors = exit code 0, though that may just be me. Depends what we think 3rd party devs might expect from empty projects with travis/other tooling. Open to thoughts on the matter.

Reply:
Guess it's fine for now!